### PR TITLE
Upgrades mcs to csc

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

File: Make.config [hunk 1]
```diff
@@ -232,6 +232,7 @@ LINKER_TOOLS_PATH=$(TOP)/external/linker
 
 MONO_PREFIX ?= /Library/Frameworks/Mono.framework/Versions/Current
 SYSTEM_MCS=$(MONO_PREFIX)/bin/mcs
+SYSTEM_CSC=$(MONO_PREFIX)/bin/mcs
```
Comment (spouliot) [inline Make.config]: `mcs` ?

File: tools/mtouch/Makefile [hunk 2]
```diff
@@ -304,12 +304,12 @@ MTOUCH_DEFINES= -define:ENABLE_BITCODE_ON_IOS
...
 
 MTOUCH_COMPILE_DEPS = $(MTOUCH_SOURCES) $(MONO_CECIL_DLL) $(MONO_CECIL_MDB_DLL) Makefile $(LINKER_RESOURCES)
-MTOUCH_COMPILE_ARGS = -unsafe $(IOS_DEFINES) -r:$(MONO_CECIL_DLL) -r:$(MONO_CECIL_MDB_DLL) -r:System.Xml.Linq.dll $(MTOUCH_SOURCES) -nowarn:0436 $(LINKER_RESOURCES:%=-resource:%) $(MTOUCH_DEFINES) -debug
+MTOUCH_COMPILE_ARGS = -unsafe $(IOS_DEFINES) -r:$(MONO_CECIL_DLL) -r:$(MONO_CECIL_MDB_DLL) -r:System.Xml.Linq.dll $(MTOUCH_SOURCES) -nowarn:0436 $(LINKER_RESOURCES:%=-resource:%) $(MTOUCH_DEFINES) -debug:portable
 
 # bin/Debug/mtouch.exe is used from mtouch.csproj when debugging
 bin/Debug/mtouch.exe: $(MTOUCH_COMPILE_DEPS)
 	$(Q) mkdir -p bin/Debug
-	$(Q) $(SYSTEM_MCS) $(MTOUCH_COMPILE_ARGS) -out:$@ -debug -d:DEV
+	$(Q_CSC) $(SYSTEM_CSC) $(MTOUCH_COMPILE_ARGS) -out:$@ -d:DEV
```
Comment (spouliot) [inline tools/mtouch/Makefile]: `debug` removed without adding `-debug:portable`

Comment (marek-safar) [inline tools/mtouch/Makefile]: Not really as MTOUCH_COMPILE_ARGS has it (you had double -debug there before)

**Comment by marek-safar** [comment]
@spouliot PTAL I'd prefer individual PRs instead of one massive csc switch PR

**Pull request merged**